---
title: "R Notebook"
output: html_notebook
---

```{r}
# Set the working directory to where the data files are located
setwd("/Users/henrique/Dropbox/Documentos/Pós-Doc/Smithsonian/BFF/results/vcf/snpden/")

# Load the required packages

library(tidyverse)
library(gdata)
library(scales)
library(ggthemes)
library(ggsci)
library(svglite)

# Read the SNP density data file
snpden <- read.table("MnigALL.MnigV2.het.1Mb.snpden.54s.july.txt", header = T)

# Read the master table data file
master <- read.table("/Users/henrique/Dropbox/Documentos/Pós-Doc/Smithsonian/BFF/bff_master_table_v4.txt", header = T)

# Join the SNP density data with the master table by individual ID
snpden.master <- right_join(snpden, master, by = "Genomics")

# Define the order of the scaffolds to be used in the visualization
target <- c('HiC_scaffold_4','HiC_scaffold_16','HiC_scaffold_3',
            'HiC_scaffold_1','HiC_scaffold_9','HiC_scaffold_17',
            'HiC_scaffold_7','HiC_scaffold_10','HiC_scaffold_6',
            'HiC_scaffold_5','HiC_scaffold_15','HiC_scaffold_19',
            'HiC_scaffold_2','HiC_scaffold_8','HiC_scaffold_18',
            'HiC_scaffold_14','HiC_scaffold_13','HiC_scaffold_12',
            'HiC_scaffold_11')

#Define the order of the chromosomes to be used in the visualization
chr <-c('chr1','chr2','chr3','chr4','chr5','chr6','chr7','chr8','chr9','chr10',
        'chr11','chr12','chr13','chr14','chr15','chr16','chr17','chr18','chrX')

# Define the chromosome length bins to be used in the visualization
#chr.len <-c('0Mbp','50Mbp','100Mbp','150Mbp','200Mbp','250Mbp','300Mbp')

# Reorder the chromosome column of the data frame according to the target order
snpden.master$CHROM <- reorder.factor(snpden.master$CHROM, new.order = target)

# Subset data from chromosomes that are not "NA"
snpden.master <-subset(snpden.master, snpden.master$CHROM!='NA')

snpden.master$groups <- cut(as.numeric(snpden.master$VARIANTS.KB), 
                            c(0,0.05,0.1,0.15,0.20,0.25,0.5,0.75,1,1.25,1.5,1.75,2,2.25,2.5,2.75,
                              3,3.25,3.5,3.75,4,4.25,4.5,4.75,5),
                            include.lowest = TRUE, labels = c("0", "0.05-0.1", "0.1-0.15", "0.15-0.2", 
                                                              "0.2-0.25", "0.25-0.5", "0.5-0.75", "0.75-1", 
                                                              "1-1.25", "1.25-1.5", "1.5-1.75", "1.75-2", 
                                                              "2-2.25", "2.25-2.5", "2.5-2.75", "2.75-3", 
                                                              "3-3.25", "3.25-3.5", "3.5-3.75", "3.75-4", 
                                                              "4-4.25", "4.25-4.5", "4.5-4.75", "4.75-5"))
snpden.master$groups[snpden.master$VARIANTS.KB == 0] <- "0"


# Rename CHROM levels
levels(snpden.master$CHROM) <-c('chr1','chr2','chr3','chr4','chr5','chr6','chr7','chr8','chr9','chr10',
                         'chr11','chr12','chr13','chr14','chr15','chr16','chr17','chr18','chrX')

snpden.master$BIN_START <- as.numeric(as.character(snpden.master$BIN_START))


for (chromosome in unique(snpden.master$CHROM)) {
  # Subset the data for the current chromosome
  snpden.chr <- subset(snpden.master, snpden.master$CHROM == chromosome)
  
  # Define title
  title<-expression(paste(italic("Mustela nigripes")))
  
  #Create ggplot object 
  snpden_plot <- snpden.chr %>% 
    mutate(BFF_pub = factor(BFF_pub, levels = c("AZ12P2","AZ14J","AZ15A","AZ15C","AZ15E","AZ15O","AZ15P","AZ15Q","AZ15W","AZ15X","AZ15Z",
                                              "SD15-04","SD15-008","SD15-006","SD14-22","SD14-021","SD14-018","SD02-111","SD02-110",
                                              "SD02-057","SD01-061","SB3226","SB9463","SB9367","SB9247","SB9026","SB8975","SB8936","SB8519",
                                              "SB8161","SB7934","SB7814","SB7235","SB7038","SB6715","SB5989","SB5777","SB5393","SB4231",
                                              "SB3876","SB3647","SB3626","SB3259","SB2766","SB731","SB663","SB295","SB256",
                                              "SB10","SB2","Meet110481","Meet107934","Mell122114","Mell122111"))) %>%
  ggplot(aes(x=BIN_START, y=1)) + 
    geom_tile(aes(fill=groups)) +
    facet_grid(BFF_pub ~ ., switch='y') +
    labs(x = 'Chromosome Length' , 
          y = 'Scaffold Number' , 
        title = expression(paste(italic("Mustela nigripes"))), 
        subtitle = paste0("Chromosome ", chromosome, " heterozygous SNP densities")) + 
    theme_minimal() +
      theme(axis.title.y=element_blank(),
          axis.text.y=element_blank(),
          axis.ticks.y=element_blank(),
          panel.grid.minor = element_blank(),
          panel.grid.major = element_blank(),
          strip.text.y.left = element_text(angle = 0, size=8),
          panel.spacing.y=unit(0.15, "lines"),
          plot.title = element_text(hjust = .5, size = 15),
          plot.subtitle = element_text(hjust = .5, size = 13, color = "gray")) +
    scale_fill_manual(values = c("#000081", "#0000f3", "#004dff", "#00b3ff", "#29ffce", 
                               "#7bff7b", "#ceff29", "#ffc600", "#ff6800", "#f30900", "brown","#800000"),
                      name = "Variants/kb",
                      labels = c("<0.05","0.05-0.10","0.10-0.15","0.15-0.20","0.20-0.25",
                                "0.25-0.50","0.50-0.75","0.75-1.0","1.0-1.25","1.25-1.5",
                                "1.5-1.75","1.75-2.0","2.0-2.25","2.25-2.5")) +  
    scale_x_continuous(name='Chromosome length', labels = c('0Mb',"50Mb", '100Mb', "150Mb", '200Mb','250Mb'),
                     breaks = c(0,50000000, 100000000, 150000000, 200000000,250000000), expand = c(0,0))



  ggsave(filename = paste0('bff_', chromosome, '.1Mb.snpden.v4.svg'), plot = snpden_plot, device = 'svg',
         dpi = 600, units = c('cm'), width = 28, height = 18, path = "plots/", bg = "white")


  ggsave(filename = paste0('bff_',chromosome, '.1Mb.snpden.v4.png'), plot = snpden_plot, device = 'png',
         dpi = 600, units = c('cm'), width = 28, height = 18, path = "plots/", bg = "white")
  

}
```


